docs/src/modules/ROOT/pages/upgrading-timefold-solver/backwards-compatibility.adoc

```diff
@@ -5,50 +5,39 @@
 
 Timefold Solver separates its API from its implementation:
 
-* **Public API**: All classes in the following package namespaces are 100% *backwards compatible* in future releases,
-especially minor and hotfix releases:
+* **Public API**: All classes under these `api` and `config` namespaces are 100% *backwards compatible* in future minor and hotfix releases:
 ** `ai.timefold.solver.core.api`
 ** `ai.timefold.solver.benchmark.api`
 ** `ai.timefold.solver.test.api`
-** `ai.timefold.solver...api*`
-* **Configuration**: The solver configuration is backwards compatible for all elements,
-except for elements that require the use of non-public API classes.
-The solver configuration is defined by the classes in the following package namespaces:
+** `ai.timefold.solver.*.api`
 ** `ai.timefold.solver.core.config`
 ** `ai.timefold.solver.benchmark.config`
 * **Implementation classes**: All other classes are _not_ backwards compatible.
 They will change in future major or minor releases,
 but probably not in hotfix releases.
 
-Backwards incompatible changes will be clearly documented in xref:upgrading-timefold-solver/upgrade-to-latest-version.adoc#manualUpgrade[the upgrade recipe].
-
-
-[NOTE]
-====
-This documentation covers some `impl` classes too.
-Those documented `impl` classes are reliable and safe to use (unless explicitly marked as experimental in this documentation),
-but we're just not entirely comfortable yet to write their signatures in stone.
-====
+Backwards incompatible changes for a new major version are clearly documented in xref:upgrading-timefold-solver/upgrade-to-latest-version.adoc#manualUpgrade[the upgrade recipe].
```

Comment:
As far as I know, the documentation doesn't mention impl classes any more, so this note can be removed.